lib/beaker/hypervisor/vmpooler.rb

```diff
@@ -246,5 +277,73 @@ def cleanup
       @logger.notify "Spent %.2f seconds cleaning up" % (Time.now - start)
     end
 
+    def add_disk(hostname, size)
+      begin
+        uri = URI.parse(@options[:pooling_api] + '/api/v1/vm/' + hostname + '/disk/' + size.to_s)
+
+        http = Net::HTTP.new(uri.host, uri.port)
+        request = Net::HTTP::Post.new(uri.request_uri)
+
+        request['X-AUTH-TOKEN'] = @credentials[:vmpooler_token]
+
+        response = http.request(request)
+      rescue RuntimeError, Errno::EINVAL, Errno::ECONNRESET, EOFError,
+             Net::HTTPBadResponse, Net::HTTPHeaderSyntaxError, *SSH_EXCEPTIONS => e
+        @logger.notify "Failed to connect to vmpooler for additional disk!"
+      end
+    end
+
+    def parse_response(response)
+      begin
+        parsed_response = JSON.parse(response.body)
+      rescue JSON::ParserError => e
+        @logger.notify "Failed to parse JSON for response from '#{host['vmhostname']}'! (failed with #{e.class})"
+      end
+    end
+
+    def disk_added?(host)
+      !host['disk'].nil?
+    end
+
+    def get_vm(hostname)
+      begin
+        uri = URI.parse(@options[:pooling_api] + '/vm/' + hostname)
+
+        http = Net::HTTP.new(uri.host, uri.port)
+        request = Net::HTTP::Get.new(uri.request_uri)
+
+        response = http.request(request)
+      rescue RuntimeError, Errno::EINVAL, Errno::ECONNRESET, EOFError,
+             Net::HTTPBadResponse, Net::HTTPHeaderSyntaxError, *SSH_EXCEPTIONS => e
+        @logger.notify "Failed to connect to vmpooler while waiting for disk!"
+      end
+
+    end
+
+    def wait_for_disk(hostname)
+
+      response = get_vm(hostname)
+      parsed = parse_response(response)
+
+      @logger.notify "Waiting for disk"
+
+      attempts = 0
+      while (!disk_added?(parsed[hostname]) && attempts < 15)
```

Comment:
This assumes only one disk could ever be added.  is this a true assumption to make?.  If so, may want to document that beaker can only add a single extra disk for VMPooler